web/src/lib/components/album-page/albums-list.svelte

```diff
@@ -0,0 +1,287 @@
+
+  onMount(async () => {
+    await removeAlbumsIfEmpty();
+  });
+
+  function showAlbumContextMenu(contextMenuDetail: OnShowContextMenuDetail, album: AlbumResponseDto): void {
+    contextMenuTargetAlbum = album;
+    contextMenuPosition = {
+      x: contextMenuDetail.x,
+      y: contextMenuDetail.y,
+    };
+  }
+
+  function closeAlbumContextMenu() {
+    contextMenuTargetAlbum = undefined;
+  }
+
+  async function handleDeleteAlbum(albumToDelete: AlbumResponseDto): Promise<void> {
+    await deleteAlbum({ id: albumToDelete.id });
+    albums = albums.filter(({ id }) => id !== albumToDelete.id);
+  }
+
+  const chooseAlbumToDelete = (album: AlbumResponseDto) => {
+    contextMenuTargetAlbum = album;
+    setAlbumToDelete();
+  };
+
+  const setAlbumToDelete = () => {
+    albumToDelete = contextMenuTargetAlbum ?? null;
+    closeAlbumContextMenu();
+  };
+
+  const handleEdit = (album: AlbumResponseDto) => {
+    selectedAlbum = { ...album };
+    shouldShowEditAlbumForm = true;
+  };
+
+  const deleteSelectedAlbum = async () => {
+    if (!albumToDelete) {
+      return;
+    }
+    try {
+      await handleDeleteAlbum(albumToDelete);
+    } catch {
+      notificationController.show({
+        message: 'Error deleting album',
+        type: NotificationType.Error,
+      });
+    } finally {
+      albumToDelete = null;
+    }
+  };
+
+  const removeAlbumsIfEmpty = async () => {
+    try {
+      for (const album of albums) {
+        if (album.assetCount == 0 && album.albumName == '') {
+          await handleDeleteAlbum(album);
+        }
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const successModifyAlbum = () => {
+    shouldShowEditAlbumForm = false;
+    notificationController.show({
+      message: 'Album infos updated',
+      type: NotificationType.Info,
+    });
+    albums[albums.findIndex((x) => x.id === selectedAlbum.id)] = selectedAlbum;
+  };
+</script>
+
+{#if shouldShowEditAlbumForm}
+  <FullScreenModal on:clickOutside={() => (shouldShowEditAlbumForm = false)}>
+    <EditAlbumForm
+      album={selectedAlbum}
+      on:editSuccess={() => successModifyAlbum()}
+      on:cancel={() => (shouldShowEditAlbumForm = false)}
+    />
+  </FullScreenModal>
+{/if}
+
+{#if albums.length > 0}
+  <!-- Album Card -->
+  {#if $albumViewSettings.view === AlbumViewMode.Cover}
+    <div class="grid grid-cols-[repeat(auto-fill,minmax(14rem,1fr))]">
+      {#each albumsFiltered as album, index (album.id)}
+        <a data-sveltekit-preload-data="hover" href="{AppRoute.ALBUMS}/{album.id}" animate:flip={{ duration: 200 }}>
+          <AlbumCard
+            preload={index < 20}
+            {album}
+            on:showalbumcontextmenu={({ detail }) => showAlbumContextMenu(detail, album)}
+          />
+        </a>
+      {/each}
+    </div>
+  {:else if $albumViewSettings.view === AlbumViewMode.List}
+    <AlbumsTable
+      bind:sortByOptions
```

Comment:
I don't think this `bind` is necessary, is it? Passing `sortByOptions` should also work just fine I think